packages/react-magma-dom/src/components/Tabs/TabsScrollSpyContainer.test.js

```diff
@@ -0,0 +1,138 @@
+import { TabScrollSpyPanel, TabsScrollSpyContainer } from '.';
+import { AndroidIcon } from 'react-magma-icons';
+
+const TEXT = 'Test Text';
+const testId = 'test-id';
+const testIdTab = `${testId}-Tab`;
+
+describe('TabsScrollSpyContainer', () => {
+  it('Should find element by testId', () => {
+    const { getByTestId } = render(<TabsScrollSpyContainer testId={testId} />);
+
+    expect(getByTestId(testId)).toBeInTheDocument();
+  });
+
+  it('Should change the active Tab on click', () => {
+    const { container, getByTestId } = render(
+      <TabsScrollSpyContainer testId={testIdTab}>
+        <TabScrollSpyPanel tabLabel={'first'}></TabScrollSpyPanel>
+        <TabScrollSpyPanel tabLabel={'second'}></TabScrollSpyPanel>
+        <TabScrollSpyPanel tabLabel={'third'}></TabScrollSpyPanel>
+      </TabsScrollSpyContainer>
+    );
+
+    const tab1 = container.querySelector('button[data-scrollspy-id="first"]');
+    const tab2 = container.querySelector('button[data-scrollspy-id="second"]');
+    const tab3 = container.querySelector('button[data-scrollspy-id="third"]');
+
+    fireEvent.click(tab1);
+
+    expect(tab1).toHaveStyleRule('color', magma.colors.primary500);
+    expect(tab2).toHaveStyleRule('color', magma.colors.neutral500);
+    expect(tab3).toHaveStyleRule('color', magma.colors.neutral500);
+
+    fireEvent.click(tab2);
+
+    expect(tab1).toHaveStyleRule('color', magma.colors.neutral500);
+    expect(tab2).toHaveStyleRule('color', magma.colors.primary500);
+    expect(tab3).toHaveStyleRule('color', magma.colors.neutral500);
+
+    fireEvent.click(tab3);
+
+    expect(tab1).toHaveStyleRule('color', magma.colors.neutral500);
+    expect(tab2).toHaveStyleRule('color', magma.colors.neutral500);
+    expect(tab3).toHaveStyleRule('color', magma.colors.primary500);
+  });
+
+  it('Should set the tabLabel', () => {
+    const { container, getByText, queryByText } = render(
+      <div style={{ height: '20px', width: '200px', overflow: 'auto' }}>
+        <TabsScrollSpyContainer activeIndex="" testId={testId}>
+          <TabScrollSpyPanel tabLabel={'first'}>
+            Content Area One
+          </TabScrollSpyPanel>
+        </TabsScrollSpyContainer>
+      </div>
+    );
+
+    expect(getByText('first')).toBeVisible();
+  });
+
+  it('Should set the icon', () => {
+    const { container, getByText, queryByText } = render(
+      <div style={{ height: '20px', width: '200px', overflow: 'auto' }}>
+        <TabsScrollSpyContainer activeIndex="" testId={testId}>
+          <TabScrollSpyPanel tabLabel={'first'} icon={<AndroidIcon />}>
+            Content Area One
+          </TabScrollSpyPanel>
+        </TabsScrollSpyContainer>
+      </div>
+    );
+
+    const tab1 = container.querySelector('button[data-scrollspy-id="first"]');
+
+    expect(tab1.firstChild.firstChild.nodeName).toBe('svg');
+  });
+
+  describe('Click events', () => {
+    it('Should allow a custom onClick function', () => {
```

Comment:
Is it valuable for the onClick prop to be on the container, or should it actually be on the panel?

Reply:
Moved to the `Tab`